tests/unit/lfp/lfp.cpp

```diff
@@ -0,0 +1,103 @@
+#define BOOST_TEST_MODULE LFPTest
+#define BOOST_TEST_MAIN
+
+#include <iostream>
+
+#include <boost/test/unit_test.hpp>
+
+#include "coreneuron/io/lfp.hpp"
+#include "coreneuron/mpi/nrnmpi.h"
+
+using namespace coreneuron;
+using namespace coreneuron::lfputils;
+
+template <typename F>
+double integral(F f, double a, double b, int n) {
+    double step = (b - a) / n;  // width of each small rectangle
+    double area = 0.0;          // signed area
+    for (int i = 0; i < n; i++) {
+        area += f(a + (i + 0.5) * step) * step;  // sum up each small rectangle
+    }
+    return area;
+}
+
+
+BOOST_AUTO_TEST_CASE(LFP_PointSource_LineSource) {
+#if NRNMPI
+    nrnmpi_init(nullptr, nullptr);
+#endif
+    double segment_length{1.0e-6};
+    double segment_start_val{1.0e-6};
+    std::array<double, 3> segment_start = std::array<double, 3>{0.0, 0.0, segment_start_val};
+    std::array<double, 3> segment_end =
+        paxpy(segment_start, 1.0, std::array<double, 3>{0.0, 0.0, segment_length});
+    double floor{1.0e-6};
+    double pi = 3.141592653589;
```

Comment:
Following the discussion between @iomaganaris and @bdelmarm : how is it going? Is it still `pi == 0`?

Reply:
As far as I could test, pi was indeed 0. @iomaganaris can you take a look again?

Reply:
I think I fixed this. Can you have look and see if you can reproduce the previous issue?

Reply:
Looks fine now. CI is passing.